ppr-ui/src/utils/utilities.ts

```diff
@@ -81,3 +83,31 @@ export const filterDuplicates = (list: Array<any>, filterBy: string) => {
     return false
   })
 }
+
+/**
+ * Deeply compares two values, supporting objects, arrays, and case-insensitive string comparison.
+ *
+ * @param {*} base - The first value to compare.
+ * @param {*} current - The second value to compare.
+ * @returns {boolean} - Returns true if the values are different, false if they are equal.
+ */
+export const deepChangesComparison = (base: BaseDataUnionIF, current: BaseDataUnionIF): boolean => {
+  // Object safety-check
+  const isObject = (value) => typeof value === 'object' && value !== null
+  // String normalization safety-check: Case Insensitive
+  const caseInsensitiveStringCompare = (val1, val2) => {
+    if (typeof val1 === 'string' && typeof val2 === 'string') {
+      return val1.toUpperCase() !== val2.toUpperCase()
```

Comment:
So we are ignoring the case? Liz was ok with that?

Reply:
It was an explicit requirement in corrections, so it should be the case here too yea. 
If they have issue, we can option this.